### ISSUE TITLE
Filter objects are not URL encoded in links on GET requests

### ISSUE THREAD

**Issue by tanj** [jfinkels/flask-restless#553]
I seem to have discovered a bug in how the filter urls are handled.

From the docs:

```
curl \
  -G \
  -H "Accept: application/vnd.api+json" \
  -d "filter[objects]=[{\"name\":\"name\",\"op\":\"like\",\"val\":\"%y%\"}]" \
  http://127.0.0.1:5000/api/person
```

this will actually fail with a 400 error. If you change the option `-d` to `--data-urlencode` it will work. However the returned json pagination links will fail because the filter will be included in a non-encoded format in those links.

**Comment by jfinkels** [comment]
@tanj Thanks for the thorough description. This is certainly annoying to come across, but it is not a bug with Flask-Restless; see the warning under the "[Operators][1]" section of the latest version of the Flask-Restless documentation. The solution is to URL-encode your strings.

[1]: http://flask-restless.readthedocs.io/en/latest/filtering.html#operators

**Comment by tanj** [comment]
Yes, I agree the solution is to URL encode my strings.

The bug is that the pagination links don't keep the URL encoding of the filter.


curl cli
```
$ curl -G -H "Accept: application/vnd.api+json" --data-urlencode "filter[objects]=[{\"name\":\"name\",\"op\":\"like\",\"val\":\"%y%\"}]" http://127.0.0.1:5000/api/person
#
# ...
#
  "links": {
    "first": "http://127.0.0.1:5000/api/person?filter[objects]=[{\"name\":\"name\",\"op\":\"like\",\"val\":\"%y%\"}]&page[number]=1&page[size]=10&filter[objects]=[{\"name\": \"name\", \"op\": \"like\", \"val\": \"%y%\"}]",
    "last": "http://127.0.0.1:5000/api/person?filter[objects]=[{\"name\":\"name\",\"op\":\"like\",\"val\":\"%y%\"}]&page[number]=1&page[size]=10&filter[objects]=[{\"name\": \"name\", \"op\": \"like\", \"val\": \"%y%\"}]",
    "next": "http://127.0.0.1:5000/api/person?filter[objects]=[{\"name\":\"name\",\"op\":\"like\",\"val\":\"%y%\"}]&page[number]=2&page[size]=10&filter[objects]=[{\"name\": \"name\", \"op\": \"like\", \"val\": \"%y%\"}]",
    "prev": "http://127.0.0.1:5000/api/person?filter[objects]=[{\"name\":\"name\",\"op\":\"like\",\"val\":\"%y%\"}]&page[number]=0&page[size]=10&filter[objects]=[{\"name\": \"name\", \"op\": \"like\", \"val\": \"%y%\"}]",
    "self": "/api/person"
  },
  "meta": {
    "total": 1
  }
}
```

quickstart.py log
```
127.0.0.1 - - [28/Jun/2016 17:32:26] "GET /api/person?filter[objects]=%5B%7B%22name%22%3A%22name%22%2C%22op%22
%3A%22like%22%2C%22val%22%3A%22%25y%25%22%7D%5D HTTP/1.1" 200 -
```

So in my opinion there is a bug in the flask-restless code. It may require a url-encoded string for the filter, but outputs a non-url-encoded string for the pagination links.

I apologize for not making that clear in the original report.

**Comment by jfinkels** [comment]
Okay, so the bug is in the links produced by Flask-Restless on fetch requests that include the `filter[objects]` query parameter. They should be URL-encoded but are not.

**Comment by jfinkels** [comment]
I have created a pull request to resolve this issue, see pull request #622.

**Comment by jfinkels** [comment]
This should be fixed in pull request #622. If any issues remain, please open a new issue with a minimal working example.
